Engine/platform/osx/acplmac.cpp

```diff
@@ -113,6 +151,31 @@ void AGSMac::ShutdownCDPlayer() {
   //cd_exit();
 }
 
+const char* AGSMac::GetAllUsersDataDirectory()
+{
+  return commonDataPath;
+}
+
+const char *AGSMac::GetUserSavedgamesDirectory()
+{
+  return libraryApplicationSupport;
+}
+
+const char *AGSMac::GetUserConfigDirectory()
+{
+  return libraryApplicationSupport;
+}
+
+const char *AGSMac::GetAppOutputDirectory()
+{
+  return commonDataPath;
+}
+
+const char *AGSMac::GetIllegalFileChars()
+{
+  return "\\/:?\"<>|*"; // keep same as Windows so we can sync.
```

Comment:
I think we may leave the function that already exists, just in case we would need it for fixing non-portable names (according to my idea I explained above).
BTW, if I remember correctly, this function was overridden only in Windows platform so far. Other platforms used default implementation, that likely is linux-compatible.

The portable file name cases deserve a separate discussion. For starters we would need to define where we do have to apply this syncronizing rule and where not.